### PR TITLE
Fix the problem that there is no cleanup script when direct gc.

### PR DESCRIPTION
**Pull request by cptbtptpbcptdtptp**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

**Comment by cptbtptpbcptdtptp** [comment]
Demo:
```
/**
 * @title Script Basic
 * @category Basic
 */
import {
  Camera,
  DirectLight,
  Entity,
  Script,
  WebGLEngine,
  Light,
  Scene
} from "@galacean/engine";

WebGLEngine.create({ canvas: "canvas" }).then((engine) => {
  engine.canvas.resizeByClientSize();
  const scene = engine.sceneManager.activeScene;
  const root = scene.createRootEntity("root");

  const cameraEntity = root.createChild("camera");
  cameraEntity.addComponent(Camera);
  cameraEntity.transform.setPosition(0, 0, 5);

  const lightEntity = root.createChild("light");
  lightEntity.addComponent(DirectLight);

  root.addComponent(TestScript);
  scene.destroy();
  engine.resourceManager.gc();
  engine.run();
});

class TestScript extends Script {
  onStart(): void {
    console.log("on start");
  }

  onUpdate(deltaTime: number): void {
    console.log("on update");
  }

  onLateUpdate(deltaTime: number): void {
    console.log("on late update");
  }

  onPhysicsUpdate(): void {
    console.log("on physics update");
  }
}

```

**Comment by GuoLei1990** [comment]
Entity destroy shouldn't dependent `gc()`

File: packages/core/src/physics/PhysicsScene.ts [hunk 1]
```diff
@@ -406,13 +406,6 @@ export class PhysicsScene {
     }
   }
 
-  /**
-   * @internal
-   */
-  _gc(): void {
-    this._colliders.garbageCollection();
-  }
-
```
Comment (GuoLei1990) [inline packages/core/src/physics/PhysicsScene.ts:415]: the same before

File: packages/core/src/lighting/LightManager.ts [hunk 2]
```diff
@@ -133,15 +133,6 @@ export class LightManager {
...
   }
 
-  /**
-   * @internal
-   */
-  _gc() {
-    this._spotLights.garbageCollection();
-    this._pointLights.garbageCollection();
-    this._directLights.garbageCollection();
-  }
```
Comment (GuoLei1990) [inline packages/core/src/lighting/LightManager.ts:143]: the same before

**Comment by GuoLei1990** [comment]
Fixed in: https://github.com/galacean/engine/pull/1724

**Pull request closed without merging**